src/util/scrollbar.ts

```diff
@@ -0,0 +1,44 @@
+import {Injectable, Inject} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+
+
+
+@Injectable()
+export class ScrollBar {
+  constructor(@Inject(DOCUMENT) private _document) {}
+
+  compensate() {
+    if (!this._isPresent()) {
+      return () => {};
+    }
+    const width = this._getWidth();
+    return this._adjustBody(width);
+  }
+
+  private _adjustBody(width: number) {
+    const {body} = this._document;
+    const userSetPadding = body.style.paddingRight;
+    const paddingAmount = parseFloat(window.getComputedStyle(body)['padding-right']);
+    body.style['padding-right'] = `${paddingAmount + width}px`;
+    return () => body.style['padding-right'] = userSetPadding;
+  }
+
+  private _isPresent(): boolean {
+    const rect = this._document.body.getBoundingClientRect();
+    return rect.left + rect.right < window.innerWidth;
+  }
+
+  private _getWidth(): number {
+    const document = this._document;
+    const {body} = document;
+
+    const measurer = document.createElement('div');
```

Comment:
By the way just realised that this is a lot more confusing to me. Unclear if it's a global `document` (wrong) or a local one.

(nitpicking, feel free to ignore)

Reply:
Yeah... again that's a matter of style. And here the context is not far to check. I made like that to avoid repetition, but also to write in a more familiar style - which is exactly what is making you confused ^^

Reply:
If you ask me I would:
* drop the `document` const (and use this._document everywhere)
* have `const documentBody = this._document.createElement('div');`

Reply:
I see the word 'document' 4 times in the snippet above, so I won't agree with the repetition argument :)

vs
```typescript
const body = this._document.body; // or const {body} = this._document;
const measurer = this._document.createElement('div')
```

I can read your version fine too though